### PR TITLE
Modify bender and runtime for Occamy compatibility

### PR DESCRIPTION
**Pull request by IveanEx**
This PR is for aligning with occamy compatibility.

Major Todos:
- [x] modify bender.yml
- [x] snRuntime

### CODE REVIEW

File: target/snitch_cluster/sw/snax/data-reshuffler/Makefile [hunk 1]
```diff
@@ -39,9 +39,20 @@ INCDIRS += $(SNRT_DIR)/src/omp
...
+
+# math.h needed by snRuntime
+
+INCDIRS += $(SNRT_DIR)/../math/arch/riscv64/
+INCDIRS += $(SNRT_DIR)/../math/arch/generic
+INCDIRS += $(SNRT_DIR)/../math/src/include
+INCDIRS += $(SNRT_DIR)/../math/src/internal
+INCDIRS += $(SNRT_DIR)/../math/include/bits
+INCDIRS += $(SNRT_DIR)/../math/include
+
```
Comment (rgantonio) [inline target/snitch_cluster/sw/snax/data-reshuffler/Makefile:51]: If these are already in `common.mk` do we have to declare them here too?

**Review by xiaoling-yi — COMMENTED** [review]
Good job! As long as all the test pass, it is ok for me!

File: target/snitch_cluster/sw/snax/data-reshuffler/Makefile (hunk 1 above)
Comment (xiaoling-yi) [reply target/snitch_cluster/sw/snax/data-reshuffler/Makefile:51]: Yes, this piece of code is repeated too much! Maybe put them into common

File: target/snitch_cluster/sw/runtime.yaml [hunk 2]
```diff
@@ -19,4 +19,4 @@ runs:
   - elf: tests/build/varargs_2.elf
   - elf: tests/build/zero_mem.elf
   - elf: tests/build/non_null_exitcode.elf
-    exit_code: 14
+    exit_code: 28
```
[the remark this answers is not in the document]
Comment (rgantonio) [reply target/snitch_cluster/sw/runtime.yaml:22]: Take out from `run.yaml` but leave in the `snitch_cluster_run.yml` thing.

File: target/snitch_cluster/sw/snax/data-reshuffler/Makefile (hunk 1 above)
Comment (rgantonio) [reply target/snitch_cluster/sw/snax/data-reshuffler/Makefile:51]: Let's copy-paste for now and do in a separate PR refactoring all the Makefile libraries 😄

File: Bender.yml [hunk 3]
```diff
@@ -24,7 +24,7 @@ dependencies:
   axi:                { git: https://github.com/pulp-platform/axi,                version:  0.39.1  }
   axi_riscv_atomics:  { git: https://github.com/pulp-platform/axi_riscv_atomics,  version:  0.6.0   }
   common_cells:       { git: https://github.com/pulp-platform/common_cells,       version:  1.28.0  }
-  FPnew:              { git: https://github.com/openhwgroup/cvfpu,                rev:      1202ca3 }  # TODO: feature branch `feature/expanding_sdotp`; get merged!
+  FPnew:              { git: https://github.com/pulp-platform/cvfpu.git,      rev:      pulp-v0.1.3 }  # TODO: feature branch `feature/expanding_sdotp`; get merged!
```
[the remark this answers is not in the document]
Comment (rgantonio) [reply Bender.yml:27]: Try the following:

1. If Bender.local works for occamy
2. Take out FP tests only and make sure this is upgraded so as long as the build is correct.

**Comment by rgantonio** [comment]
For banshee and RTL fails, need the `team_global` test to pass. Check if the `non_null_exit` test passes with the exit code of 126.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -156,8 +158,7 @@ jobs:
         working-directory: target/snitch_cluster
         run: |-
           ./run.py --simulator verilator \
-          sw/runtime.yaml sw/standard-fp.yaml \
-          sw/snax-streamer-gemm-conv-run.yaml -j
+          sw/runtime.yaml sw/snax-streamer-gemm-conv-run.yaml -j
```
Comment (rgantonio) [inline .github/workflows/ci.yml:161]: Maybe return the `sw/standard-fp.yaml` since it now passes with our tests right? 😄

File: Bender.local [hunk 5]
```diff
@@ -5,3 +5,6 @@
 overrides:
     # Some of our dependencies have false conflicts with our new AXI version; force our version.
     axi: {git: https://github.com/pulp-platform/axi.git, version: 0.39.0}
+    # Original snitch cluster uses the old FPnew, while occamy uses the new FPnew
+    # FPnew: { git: https://github.com/openhwgroup/cvfpu,    rev: 1202ca3 }
+
```
Comment (rgantonio) [inline Bender.local:10]: Erase! haha and don't forget new line.

File: sw/tests/team_global.c
Comment (rgantonio) [inline sw/tests/team_global.c:1]: Just for documentation purposes, we discussed this and found out that the original Snitch cluster has this update:

https://github.com/pulp-platform/snitch_cluster/blob/e90dceb6a6c9bdb6bb5d1502b62b920242be9fa8/sw/tests/team_global.c#L8-L23

File: target/snitch_cluster/sw/custom-fp.yaml
Comment (rgantonio) [inline target/snitch_cluster/sw/custom-fp.yaml:1]: For documentation purposes please make an issue of this as a separate PR to fix the Snitch to accommodate the hard changes with the new FPU.

File: target/snitch_cluster/sw/snitch-cluster-runtime.yaml
Comment (rgantonio) [inline target/snitch_cluster/sw/snitch-cluster-runtime.yaml:1]: Also for documentation purposes the Snitch cluster has these changes too:

https://github.com/pulp-platform/snitch_cluster/blob/e90dceb6a6c9bdb6bb5d1502b62b920242be9fa8/target/snitch_cluster/sw/run.yaml#L72-L73

**Review by rgantonio — APPROVED** [review]
Great!

File: target/snitch_cluster/sw/custom-fp.yaml
Comment (rgantonio) [reply target/snitch_cluster/sw/custom-fp.yaml:1]: Mentioned in issue #154

File: .github/workflows/ci.yml (hunk 4 above)
Comment (IveanEx) [reply .github/workflows/ci.yml:161]: Only the original snitch cluster supports FP, and all scalar tests are not removed.

Comment (rgantonio) [reply .github/workflows/ci.yml:161]: Ah i didn't see it was for Gemm LOL okay okay
